dvc/repo/ls_url.py

```diff
@@ -0,0 +1,39 @@
+from dvc.exceptions import URLMissingError
+from dvc.fs import get_cloud_fs
+
+
+def parse_external_url(url):
+    fs_cls, fs_config, fs_path = get_cloud_fs(None, url=url)
+    fs = fs_cls(**fs_config)
+    return fs, fs_path
+
+
+def ls_url(url, recursive=False):
+    fs, fs_path = parse_external_url(url)
+    try:
+        return _ls_url(fs, fs_path, recursive=recursive)
+    except FileNotFoundError as exc:
+        raise URLMissingError(url) from exc
```

Comment:
This might hide some odd `walk` errors. That's why we only wrap info in `dvc ls`.